src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -163,14 +170,13 @@ private TypeSymbol GetBestType(ImmutableArray<TypeSymbol> types, ref HashSet<Dia
 
             // We have actually only determined that every type *after* best was worse. Now check
             // that every type *before* best was also worse.
-            foreach (var type in types)
+            for (int i = 0; i < bestIndex; i++)
```

Comment:
Could we loop while `i < bestIndex - 1` and remove the `(object)better != (object)best` check?